python/graphstorm/sagemaker/sagemaker_partition.py

```diff
@@ -255,7 +255,7 @@ def run_partition(job_config: PartitionJobConfig):
     for key, val in os.environ.items():
         logging.debug("%s: %s", key, val)
 
-    leader_addr = socket.gethostbyname('algo-1')
+    leader_addr = socket.gethostbyname(sorted(hosts)[0])
```

Comment:
It doesn't really matter which one we choose as long as all instances choose the same. This guarantees this will happen, previously `'algo-1'` was causing errors in local mode because instances are named differently there.